src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -281,24 +281,41 @@ class WP_HTML_Processor extends WP_HTML_Tag_Processor {
 	 *
 	 * ## Current HTML Support
 	 *
-	 *  - The only supported context is `<body>`, which is the default value.
 	 *  - The only supported document encoding is `UTF-8`, which is the default value.
 	 *
+	 * @todo Verify that creating a fragment in self-contained elements works.
+	 *
 	 * @since 6.4.0
 	 * @since 6.6.0 Returns `static` instead of `self` so it can create subclass instances.
+	 * @since 6.7.0 Can create fragment in any context.
 	 *
 	 * @param string $html     Input HTML fragment to process.
 	 * @param string $context  Context element for the fragment, must be default of `<body>`.
 	 * @param string $encoding Text encoding of the document; must be default of 'UTF-8'.
 	 * @return static|null The created processor if successful, otherwise null.
 	 */
 	public static function create_fragment( $html, $context = '<body>', $encoding = 'UTF-8' ) {
-		if ( '<body>' !== $context || 'UTF-8' !== $encoding ) {
+		if ( 'UTF-8' !== $encoding ) {
+			return null;
+		}
+
+		$context_processor = new WP_HTML_Tag_Processor( $context );
+		if ( ! $context_processor->next_token() || '#tag' !== $context_processor->get_token_type() ) {
+			return null;
+		}
+
+		$context_tag        = $context_processor->get_tag();
+		$context_attributes = array();
+		foreach ( $context_processor->get_attribute_names_with_prefix( '' ) as $name ) {
```

Comment:
Having an empty prefix here looks strange.
If it was my code I'd either have the prefix optional/default to empty or add a `get_attributes_names()` which would do exactly the same as `get_attributes_names_with_prefix( '' )`, might be redundant in a way, but for me it would look a little more readable.

Reply:
thanks @apermo. if this looks strange, it's supposed to 🙃 
when we added this method we wanted it to be a bit awkward in the hopes of communicating that it involves additional costs. we wanted the default behavior to be looking for a subset of attributes (such as `get_attribute_names_with_prefix( 'data-' )`) so that the API itself guides people to learn in the safest most performant manner.